server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java

```diff
@@ -182,6 +183,22 @@ static SqlNode renameColumns(@Nullable SqlNodeList columnList,
         null, null, null, null, null);
   }
 
+  static void erase(SqlIdentifier name, CalcitePrepare.Context context) {
+    // Directly clearing data is more efficient than executing SQL
+    final Pair<CalciteSchema, String> pair = schema(context, true, name);
+    final CalciteSchema calciteSchema = pair.left;
+    final String tblName = pair.right;
+    final Table table = calciteSchema
+        .getTable(tblName, context.config().caseSensitive())
+        .getTable();
+    if (table instanceof MutableArrayTable) {
+      MutableArrayTable mutableArrayTable = (MutableArrayTable) table;
+      mutableArrayTable.rows.clear();
+    } else {
+      throw new UnsupportedOperationException("Only MutableArrayTable support truncate");
```

Comment:
May I ask why only MutableArrayTable is supported? ^_^

Reply:
- The DELETE statement cannot delete data in the server module.
   I initially thought it was a [bug](https://issues.apache.org/jira/browse/CALCITE-5796). But after analysis, it  is limitation of the server module.
- `MutableArrayTable` is enough, because `CREATE` table all are `MutableArrayTable`. Here is code link: https://github.com/apache/calcite/blob/301f770a66ef3d26c9d4e8d274eae0f92a819e00/server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java#L515

Reply:
OK, got it. Maybe some java doc is better for other developers, WDYT?

Reply:
Make sense, I added some explanations.